### PR TITLE
Implement write_{8,word_32} for GDBCore

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: humility-core/src/core/gdb.rs [hunk 1]
```diff
@@ -412,16 +414,24 @@ impl Core for GDBCore {
         ))
     }
 
-    fn write_word_32(&mut self, _addr: u32, _data: u32) -> Result<()> {
-        Err(anyhow!(
-            "{} GDB target does not support modifying state", self.server
-        ))
+    fn write_word_32(&mut self, addr: u32, data: u32) -> Result<()> {
+        let bytes: [u8; 4] = unsafe { transmute(data.to_le()) };
```
Comment (beezow) [inline humility-core/src/core/gdb.rs]: Do you need the unsafe transmute.  Can you just do `data.to_le_bytes()`? https://play.rust-lang.org/?version=beta&mode=debug&edition=2015&gist=7e64bcd8d99a1628ac9b5f9666d8e3a9

File: humility-core/Cargo.toml [hunk 2]
```diff
@@ -35,6 +35,7 @@ strum_macros = "0.24.1"
 clap = "3.0.12"
 roxmltree = "0.15"
 xmlparser = "0.13"
+hex = "0.4.3"
```
Comment (beezow) [inline humility-core/Cargo.toml:38]: You'll have to update the nix hash within `humility.nix` with this change.  The CI _should_ tell you what the new hash is.  I need to transition to [naersk](https://github.com/nix-community/naersk) at some point so manually changing the hash is not required.

**Review by beezow — CHANGES_REQUESTED** [review]
Overall looks good. What did you do to test this?

File: humility-core/src/core/gdb.rs (hunk 1 above)
Comment (ghost) [reply humility-core/src/core/gdb.rs]: Thanks, data.to_le_bytes() works.

File: humility-core/Cargo.toml (hunk 2 above)
Comment (ghost) [reply humility-core/Cargo.toml:38]: Hash has been updated in branch update.

**Comment by ghost** [comment]
> Overall looks good. What did you do to test this?

I used `humility jefe` to set the task disposition of the 'ping' task for demo-rv64-qemu-virt as this makes use of write_word_32(). Setting the disposition to '--hold' causes 'ping' to go into 'FAULT' state on it's 'Illegal instruction' which can be verified by running 'humility tasks ping':

```
humility: attached via QEMU GDB server
system time = 54468345
ID TASK                       GEN PRI STATE
 2 ping                      3751   4 FAULT: illegal instruction (was: ready)
```

**Review by beezow — APPROVED** [review]

**Pull request merged**